### ISSUE TITLE
Resharing is blocked unless all the nodes are online

### ISSUE THREAD

**Issue by volovyks** [sig-net/mpc#11]
### Description
When a new participant joins, we require all the participants to be online.
(needs online)

**Comment by volovyks** [comment]
This problem was purely described.
The expected behavior is:
- When the node joins, all old participants and the new participant must be online
- When the node is kicked, all REMAINING participants must be online

As a result, blocked resharing is the expected behavior when the node joins.

This logic is based on the fact that we can add or remove only one participant at a time.
cc @scafuro

**Comment by ChaoticTempest** [comment]
We don't need all old participants to be online, we just need all new participants to be.

**Comment by volovyks** [comment]
Yes, but in case we add a node, technically, we need all the old participants.

**Comment by scafuro** [comment]
@volovyks yes we need all old participants in they wish to still participate in the future. 

Let me explain: In principle we do not need all parties since any set of t parties is sufficient to perform a correct re-sharing. However, all parties who wish to participate in the protocol going forward must be present during the re-sharing.
Example: Suppose there are 8 existing participants with a threshold of 5, and 2 new participants want to join. While 5 parties are sufficient to perform the re-sharing, the 3 old participants who were offline during the process will lose their ability to participate, because their previous shares become obsolete. Discarding old shares upon each re-sharing is crucial for maintaining security.
